src/position.ts

```diff
@@ -170,6 +207,47 @@ class PositionWithRunner extends Position {
     await Promise.all([collateral, debt]);
   }
 
+  /**
+   * Fetches the list of transactions of the position.
+   * @returns The list of transactions.
+   */
+  public async getTransactions(): Promise<PositionTransaction[]> {
+    const query = gql`
+      query GetTransactions($ownerAddress: String!) {
+        position(id: $ownerAddress) {
+          transactions(orderBy: timestamp, orderDirection: desc) {
+            id
+            type
+            collateralToken
+            collateralChange
+            debtChange
+            timestamp
+          }
+        }
+      }
+    `;
+
+    const userAddress = await this.getUserAddress();
+    const response = await request<PositionTransactionsQuery>(SUBGRAPH_ENDPOINT_URL, query, {
+      ownerAddress: userAddress.toLowerCase(),
+    });
+
+    if (!response.position?.transactions) {
+      return [];
+    }
+
+    return response.position.transactions.map(
+      transaction =>
+        ({
+          ...transaction,
+          collateralToken: RaftConfig.getTokenTicker(transaction.collateralToken) ?? '',
+          collateralChange: Decimal.parse(transaction.collateralChange, Decimal.PRECISION),
+          debtChange: Decimal.parse(transaction.debtChange, Decimal.PRECISION),
+          timestamp: new Date(Number(transaction.timestamp) * 1000),
+        } as PositionTransaction),
```

Comment:
Is this cast needed?

Reply:
Yes, since timestamp is string

Reply:
Timestamp is string in response data, but it's converted to date. To it should be fine even without `as PositionTransaction`

Reply:
Oh that cast